packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_base.py

```diff
@@ -193,26 +194,51 @@ def transport(self) -> str:
         return self._transport
 
     async def __aenter__(self):
-        if self._exit_stack:
+        if self._context_manager:
```

Comment:
AsyncExitStack is still needed to handled mcp_client function group in its current form. @sjarmak can you pls migrate your agent from mcp_tool_wrapper to mcp_client and let us know if this is an issue there?